src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_GetLatestDiagnosticsForSpan.cs

```diff
@@ -107,18 +109,137 @@ private class LatestDiagnosticsForSpanGetter
                 }
             }
 
+            private async Task<bool> TryGetSyntaxAndSemanticDiagnosticsAsync(StateSet stateSet, VersionStamp textVersion, VersionStamp syntaxVersion, VersionStamp semanticVersion)
+            {
+                // unfortunately, we need to special case compiler diagnostic analyzer so that 
+                // we can do span based analysis even though we implemented it as semantic model analysis
```

Comment:
yep. this is only for compiler. it doesnt even attempt to use compiler diagnostic analyzer. basically, it bypass diagnostic analyzer engine completely, and go directly compiler and ask diagnostics.

like @mavasani said, our current diagnostic analyzer design never designed to support spans the ground up (since design was led by compiler side and instrumentality is not an issue for compiler) it was kind of hacked around in IDE to support instrumentality. 

so, no diagnostic analyzer will do span based analysis. period.